releasenotes/notes/cleanup-curve-analysis-96d7ff706cae5b4e.yaml

```diff
@@ -0,0 +1,31 @@
+---
+developer:
+  - |
+    :class:`CurveAnalysis` is now a subclass of :class:`BaseCurveAnalysis`.
+    This change has been made to introduce multi-grouped curve analysis,
+    see qiskit-experiments/#737 for details.
+    
+    According to this update, several protected methods that are
+    originally designed to be overridden by subclass have been deprecated or upgraded.
+    Check the following list of changes you may need to upgrade your analysis subclass.
+    
+    - :meth:`CurveAnalysis._generate_fit_guesses`
+    
+    The signature of method has been upgraded. Now this method should be called with
```

Comment:
This is something that should go in upgrade note. eg "The signature of several internal methods of CurveAnalysis used by analysis subclasses have been changed <details>"